feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -84,17 +85,17 @@ fun ShowTestableToast(
             .size(20.dp)
             .testTag(toastMessage.testTag)
     ) {
-        // prevents toast from being spammed
-        if (!toastShownStatus.value) {
+        // checking toastShownStatus prevents toast visual from being spammed
```

Comment:
It seems like we're fighting against compose here a bit. Since `Toast` is not `@Composable`, we probably want to put it into a side-effect. And since side-effects will only run when a key changes, we can use `toastMessage` as a key:
```
@Composable
fun TestableToast(
    modifier: Modifier = Modifier,
    toastMessage: ToastMessage,
    onToastShown: () -> Unit
) {
    Box(
        // box seems to need to have some size to be detected by UiAutomator
        modifier = modifier
            .size(20.dp)
            .testTag(toastMessage.testTag)
    )

    val context = LocalContext.current
    LaunchedEffect(toastMessage) {
        if (toastMessage.shouldShowToast) {
            Toast.makeText(
                context,
                context.getText(toastMessage.stringResource),
                toastMessage.toastLength
            ).show()
        }
        onToastShown()
    }
}
```